pkg/config/supportedcontrollers.go

```diff
@@ -0,0 +1,105 @@
+package config
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+const (
+	// Unsupported is the default enum for non-defined controller types
+	Unsupported SupportedController = iota
+	// Deployments are a supported controller for scanning pod specs
+	Deployments
+	// StatefulSets are a supported controller for scanning pod specs
+	StatefulSets
+	// DaemonSets are a supported controller for scanning pod specs
+	DaemonSets
+	// Jobs are a supported controller for scanning pod specs
+	Jobs
+	// CronJobs are a supported controller for scanning pod specs
+	CronJobs
+)
+
+// ControllerStrings are strongly ordered to match the SupportedController enum
+var ControllerStrings = []string{
+	"Unsupported",
+	"Deployments",
+	"StatefulSets",
+	"DaemonSets",
+	"Jobs",
+	"CronJobs",
+}
+
+// stringLookupForSupportedControllers is the list of lowercase singular and plural strings for string to enum lookup
+var stringLookupForSupportedControllers = map[string]SupportedController{
+	"deployment":   Deployments,
+	"deployments":  Deployments,
+	"statefulset":  StatefulSets,
+	"statefulsets": StatefulSets,
+	"daemonset":    DaemonSets,
+	"daemonsets":   DaemonSets,
+	"job":          Jobs,
+	"jobs":         Jobs,
+	"cronjob":      CronJobs,
+	"cronjobs":     CronJobs,
+}
+
+// SupportedController is a constant item of a controller that is supported for scanning pod specs
+type SupportedController int
+
+// String returns the string name for a given SupportedController enum
+func (s SupportedController) String() string {
+	return ControllerStrings[s]
+}
+
+// MarshalJSON manages writing the enum into json data or error on unsupported value
+func (s SupportedController) MarshalJSON() ([]byte, error) {
+	if s == Unsupported {
+		return []byte{}, fmt.Errorf("Unsupported is not a valid Supported Controller")
+	}
+	buffer := bytes.NewBufferString(`"`)
+	buffer.WriteString(s.String())
+	buffer.WriteString(`"`)
+	return buffer.Bytes(), nil
+}
+
+// UnmarshalJSON handles reading json data into enum
+func (s *SupportedController) UnmarshalJSON(b []byte) error {
+	var j string
+	err := json.Unmarshal(b, &j)
+	if err != nil {
+		return err
+	}
+
+	*s, err = GetSupportedControllerFromString(j)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetSupportedControllerFromString fuzzy matches a string with a SupportedController Enum
+func GetSupportedControllerFromString(str string) (SupportedController, error) {
+	lowerStr := strings.ToLower(str)
+	if stringLookupForSupportedControllers[lowerStr] == Unsupported {
+		return 0, fmt.Errorf("Value ('%v') in configuration was not found in Supported Controllers: (%v)", str, strings.Join(ControllerStrings, ","))
+	}
+	return stringLookupForSupportedControllers[lowerStr], nil
```

Comment:
Do we need to check if `lowerStr` is in the map?

Reply:
Hmm yes it would appear so. I'll add some tests around this and see what the default behavior is for nil retirns.

Reply:
So this is a weird language behavior - i find it strange but it's supposedly supported :man_shrugging: 

https://blog.golang.org/go-maps-in-action

TL;DR; 
if you have a `map[string]int` and you call a key that doesn't exist - you get the `zero value` default of the _value_ type. So in our case `map[string]SupportedController` -- `SupportedController is an enum which has a value of `int`, and in this case `int`s "`zero value`" is 0. (for strings it's `""` and for `interfaces`, `func`, `struct`, you can get `nil` as the zero value)